cmd/kube-apiserver/app/options/options.go

```diff
@@ -136,9 +135,10 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 		"DEPRECATED: kubelet port.")
 	fs.MarkDeprecated("kubelet-port", "kubelet-port is deprecated and will be removed.")
 
-	fs.UintVar(&s.KubeletConfig.ReadOnlyPort, "kubelet-read-only-port", s.KubeletConfig.ReadOnlyPort,
-		"DEPRECATED: kubelet read only port.")
-	fs.MarkDeprecated("kubelet-read-only-port", "kubelet-read-only-port is deprecated and will be removed.")
+	var unused uint
+	fs.UintVar(&unused, "kubelet-read-only-port", 0,
+		"DEPRECATED: kubelet read only port. Value ignored even if set.")
+	fs.MarkDeprecated("kubelet-read-only-port", "kubelet-read-only-port is deprecated, ignored and will be removed.")
```

Comment:
This flag [seem to be deprecated](https://github.com/kubernetes/kubernetes/commit/d0997a3d1fc3#diff-3149c6ede53e3a658bbcf11f1456f2ac955ae49db5ab66ee9e13f883488349d3R155-R158) long time ago. Should we simply remove it?

Reply:
Probably - I can't imagine anyone using it, since it never worked. I was being conservative.